File: desktop/src/types/pieces.ts
import type { EnhancedMedia } from "./importing";

export type PieceGenerationResponse = {
  job_id: number;
  total_available_media: number;
  pieces_created: number;
  pieces_skipped: number;
};

export type ContentPieceMedia = {
  id: number;
  piece_id: number;
  original_media_id: number | null;
  enhanced_media_id: number | null;
  position: number;
  role: string | null;
  notes: string | null;
  enhanced_media: EnhancedMedia | null;
};

export type ContentPiece = {
  id: number;
  event_id: number;
  piece_type: string;
  title: string;
  purpose: string | null;
  target_platform: string | null;
  aspect_ratio: string | null;
  output_path: string | null;
  status: string;
  metadata_json: string | null;
  created_at: string;
  updated_at: string;
  approved_at: string | null;
  rejected_at: string | null;
  media_items: ContentPieceMedia[];
};

export type ContentPieceListResponse = {
  items: ContentPiece[];
};

export type ContentPieceUpdate = {
  title?: string;
  purpose?: string | null;
  target_platform?: string | null;
  aspect_ratio?: string | null;
  status?: string;
  media_item_order?: number[];
};

File: backend/app/schemas/pieces.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field

from app.schemas.importing import EnhancedMediaResponse


class PieceGenerationResponse(BaseModel):
    job_id: int
    total_available_media: int
    pieces_created: int
    pieces_skipped: int


class ContentPieceMediaResponse(BaseModel):
    id: int
    piece_id: int
    original_media_id: int | None
    enhanced_media_id: int | None
    position: int
    role: str | None
    notes: str | None
    enhanced_media: EnhancedMediaResponse | None


class ContentPieceResponse(BaseModel):
    id: int
    event_id: int
    piece_type: str
    title: str
    purpose: str | None
    target_platform: str | None
    aspect_ratio: str | None
    output_path: str | None
    status: str
    metadata_json: str | None
    created_at: datetime
    updated_at: datetime
    approved_at: datetime | None
    rejected_at: datetime | None
    media_items: list[ContentPieceMediaResponse]


class ContentPieceListResponse(BaseModel):
    items: list[ContentPieceResponse]


class ContentPieceUpdate(BaseModel):
    title: str | None = Field(default=None, min_length=1, max_length=220)
    purpose: str | None = Field(default=None, max_length=160)
    target_platform: str | None = Field(default=None, max_length=80)
    aspect_ratio: str | None = Field(default=None, max_length=30)
    status: str | None = Field(default=None, min_length=1, max_length=40)
    media_item_order: list[int] | None = None
